compiler/luci/service/src/Nodes/CircleFullyConnected.cpp

```diff
@@ -36,4 +40,70 @@ luci::CircleNode *CloneNodeLet<CN::DEF>::visit(const luci::CircleFullyConnected
   return cloned;
 }
 
+namespace sinf
+{
+
+loco::TensorShape Algorithm::visit(const luci::CircleFullyConnected *node)
+{
+  auto input_shape = circle_shape(loco::must_cast<CircleNode *>(node->input()));
+  auto weights_shape = circle_shape(loco::must_cast<CircleNode *>(node->weights()));
+
+  loco::TensorShape out_shape;
+
+  // NOTE Some recipes in some repositories are using rank 4 input for FullyConnected.
+  //      Until they are all fixed, disable following assert.
+  // TODO Enable following assert after related fixes are applied
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L194
+  // LUCI_ASSERT(input_shape.rank() == 2 || input_shape.rank() == 3,
+  //             "Input rank of FullyConnected should be 2 or 3");
+
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L225
+  LUCI_ASSERT(weights_shape.rank() == 2, "Weights of FullyConnected should be 2");
+
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L353-L367
+  if (node->keep_num_dims())
+  {
+    out_shape.rank(input_shape.rank());
+    for (uint32_t i = 0; i < input_shape.rank(); ++i)
+      out_shape.dim(i) = input_shape.dim(i);
+    out_shape.dim(out_shape.rank() - 1) = weights_shape.dim(0);
+  }
+  else
+  {
+    uint32_t input_size = 1;
+    bool is_dynamic_shape = false;
+    for (uint32_t i = 0; i < input_shape.rank(); i++)
+    {
+      if (not input_shape.dim(i).known() && i != input_shape.rank() - 1)
+      {
+        is_dynamic_shape = true;
+        break;
+      }
+      input_size = input_size * (input_shape.dim(i).known() ? input_shape.dim(i).value()
+                                                            : weights_shape.dim(1).value());
+    }
```

Comment:
```suggestion
    bool is_dynamic_shape = false;
    for (uint32_t i = 0; i < input_shape.rank()-1; i++)
    {
      if (not input_shape.dim(i).known())
      {
        is_dynamic_shape = true;
        break;
      }
    }
```

This might be a personal preference, So, I don't force you to fix it. 
It is just my recommendation and you don't have to follow it. 

How about removing `input_size` calculating part in this for loop? 
IMHO, Each for loop with only one rules helps to write the easy code.

Reply:
> you don't have to follow it.

If you want to keep current style, plz let me know. 
I'd like to follow your opinion :)